### ISSUE TITLE
Unable to install PHP on Linux

### ISSUE THREAD

**Issue by ChrisWiegman** [Homebrew/linuxbrew-core#21533]
# Bug report

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

`brew install php` on Pop!_OS 20.04

## What happened (include command output)

output link from gist-logs:
https://gist.github.com/ChrisWiegman/e61272e5af0dd9d8897148ea373e45cf

<details>
  <summary>Command output</summary>
  <pre>

libiconv: is not needed on Linux, where iconv.h is provided by glibc
Error: An unsatisfied requirement failed this build.

  </pre>
</details>

## What you expected to happen

php should be installed

## Step-by-step reproduction instructions (by running `brew install` commands)

brew install php

**Comment by ChrisWiegman** [comment]
I should add that `brew deps --tree libiconv` doesn't show the dep at all. Seems strange as the error is clearly coming from the dep file for libiconv (see https://github.com/Homebrew/linuxbrew-core/blob/4d62da1884f9f60948ea97a5896b44a6d0d197e1/Formula/libiconv.rb#L5)

**Comment by ChrisWiegman** [comment]
Tracking down dependencies 1x1 I've been able to install all dependencies sans `curl-openssl` and `openldap` both of which trigger the error above. Manually installing remaining dependencies and then running `brew install php --ignore-dependencies`. 

I'm afraid I've hit the limit of what my own knowledge of Homebrew can help dig down to this morning. Hopefully this is enough to help reproduce the issue.

**Comment by ChrisWiegman** [comment]
Final output of `brew doctor`, with a working PHP, is as follows:

```brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are missing dependencies.
You should `brew install` the missing dependencies:
  brew install curl-openssl libidn libmetalink libssh2 openldap rtmpdump zstd

Run `brew missing` for more details.
```

**Comment by iMichka** [comment]
I am working on a fix here: #21539
